### ISSUE TITLE
[Bug]: global `a11y: { test: 'error' }` from `preview.ts` is ignored

### ISSUE THREAD

**Issue by artaommahe** [storybookjs/storybook#32372]
### Describe the bug

Setting `a11y: { test: 'error' }` globally in `preview.ts` doesn't work and `a11yParameter` [here](https://github.com/storybookjs/storybook/blob/a13bee6a5ca706df20b3fcf14bd31f842e4717f8/code/addons/a11y/src/preview.tsx#L18) contains default `test` value from [this object](https://github.com/storybookjs/storybook/blob/a13bee6a5ca706df20b3fcf14bd31f842e4717f8/code/addons/a11y/src/preview.tsx#L101) (verified by changing it to some random string directly in `node_modules` and debugging `a11yParameter` value).

### Reproduction link

https://github.com/artaommahe/test-storybook-a11y-tests

### Reproduction steps

- clone https://github.com/artaommahe/test-storybook-a11y-tests
- `npm install`
- `npm run test`
- see no errors despite having `a11y: { test: "error" }` in `.storybook/preview.ts`
- uncomment `a11y: { test: "error" }` in story file `src/stories/Button.stories.ts`
- `npm run test`
- see expected error

### System

```bash
System:
    OS: macOS 15.6.1
    CPU: (10) arm64 Apple M2 Pro
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 24.1.0 - ~/.nvm/versions/node/v24.1.0/bin/node
    npm: 11.3.0 - ~/.nvm/versions/node/v24.1.0/bin/npm <----- active
  Browsers:
    Chrome: 139.0.7258.155
    Safari: 18.6
  npmPackages:
    @storybook/addon-a11y: ^9.1.3 => 9.1.3 
    @storybook/addon-vitest: ^9.1.3 => 9.1.3 
    @storybook/react-vite: ^9.1.3 => 9.1.3 
    eslint-plugin-storybook: ^9.1.3 => 9.1.3 
    storybook: ^9.1.3 => 9.1.3
```

**Comment by yannbf** [comment]
Hey there @artaommahe the issue is related to your [set up file](https://github.com/artaommahe/test-storybook-a11y-tests/blob/main/.storybook/vitest.setup.ts#L7):

```
setProjectAnnotations([projectAnnotations, addonA11y])
```

The a11y annotations have to be put first. This is automatically set up for instance when you install the vitest addon e.g. `npx storybook add @storybook/addon-vitest`. The order is then as follows:
```
setProjectAnnotations([addonA11y, projectAnnotations])
```

Then your tests behave as expected.
